### PR TITLE
[Merged by Bors] - feat(Order/WithBot): add some lemmas

### PR DESCRIPTION
**Pull request by astrainfinita**
The `WithTop` version of the last 3 lemmas already exists.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/WithBot.lean [hunk 1]
```diff
@@ -225,6 +225,11 @@ theorem not_coe_le_bot (a : α) : ¬(a : WithBot α) ≤ ⊥ := fun h =>
   Option.not_mem_none _ hb
 #align with_bot.not_coe_le_bot WithBot.not_coe_le_bot
 
+@[simp]
+theorem le_bot_iff : ∀ {a : WithBot α}, a ≤ ⊥ ↔ a = ⊥
+  | (a : α) => by simp [not_coe_le_bot _]
+  | ⊥ => by simp
```
Comment (eric-wieser) [inline Mathlib/Order/WithBot.lean:231]: Isn't this the same as https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/BoundedOrder.html#le_bot_iff?

File: Mathlib/Order/WithBot.lean [hunk 2]
```diff
@@ -1168,6 +1178,19 @@ theorem map_le_iff [Preorder α] [Preorder β] (f : α → β) (a b : WithTop α
   simp [mono_iff]
 #align with_top.map_le_iff WithTop.map_le_iff
 
+theorem coe_untop'_le [Preorder α] : ∀ (a : WithTop α) (b : α), a.untop' b ≤ a
+  | (a : α), _ => le_rfl
+  | ⊤, _ => le_top
+
+theorem le_untop'_top_iff [LE α] [OrderTop α] {a : WithTop α} {b : α} :
+    b ≤ a.untop' ⊤ ↔ b ≤ a := by
```
Comment (eric-wieser) [inline Mathlib/Order/WithBot.lean:1186]: This is a special case of `(h : a ≤ c) : b ≤ a.untop' c ↔ b ≤ a` right?

File: Mathlib/Order/WithBot.lean (hunk 1 above)
Comment (negiizhao) [reply Mathlib/Order/WithBot.lean:231]: This does not need a `PartialOrder`.

Comment (eric-wieser) [reply Mathlib/Order/WithBot.lean:231]: Can you add a docstring that explains that?

File: Mathlib/Order/WithBot.lean [hunk 3]
```diff
@@ -225,6 +225,12 @@ theorem not_coe_le_bot (a : α) : ¬(a : WithBot α) ≤ ⊥ := fun h =>
   Option.not_mem_none _ hb
 #align with_bot.not_coe_le_bot WithBot.not_coe_le_bot
 
+/-- There is a general version `le_bot_iff`, but this lemma does not require a `PartialOrder`. -/
+@[simp]
+theorem le_bot_iff : ∀ {a : WithBot α}, a ≤ ⊥ ↔ a = ⊥
```
Comment (YaelDillies) [inline Mathlib/Order/WithBot.lean:230]:
```suggestion
protected theorem le_bot_iff : ∀ {a : WithBot α}, a ≤ ⊥ ↔ a = ⊥
```

File: Mathlib/Order/WithBot.lean [hunk 4]
```diff
@@ -392,16 +398,17 @@ theorem le_coe_unbot' [Preorder α] : ∀ (a : WithBot α) (b : α), a ≤ a.unb
   | ⊥, _ => bot_le
 #align with_bot.le_coe_unbot' WithBot.le_coe_unbot'
 
-theorem unbot'_bot_le_iff [LE α] [OrderBot α] {a : WithBot α} {b : α} :
-    a.unbot' ⊥ ≤ b ↔ a ≤ b := by
-  cases a <;> simp [none_eq_bot, some_eq_coe]
-#align with_bot.unbot'_bot_le_iff WithBot.unbot'_bot_le_iff
+theorem unbot'_le_iff [LE α] {a : WithBot α} {b c : α} (h : a ≠ ⊥ ∨ b ≤ c) :
```
Comment (YaelDillies) [inline Mathlib/Order/WithBot.lean:401]: What about this?
```suggestion
theorem unbot'_le_iff [LE α] {a : WithBot α} {b c : α} (h : a = ⊥ → b ≤ c) :
```

File: Mathlib/Order/WithBot.lean [hunk 5]
```diff
@@ -394,16 +400,17 @@ theorem le_coe_unbot' [Preorder α] : ∀ (a : WithBot α) (b : α), a ≤ a.unb
...
 
-theorem unbot'_bot_le_iff [LE α] [OrderBot α] {a : WithBot α} {b : α} :
-    a.unbot' ⊥ ≤ b ↔ a ≤ b := by
-  cases a <;> simp [none_eq_bot, some_eq_coe]
-#align with_bot.unbot'_bot_le_iff WithBot.unbot'_bot_le_iff
+theorem unbot'_le_iff [LE α] {a : WithBot α} {b c : α} (h : a = ⊥ → b ≤ c) :
+    a.unbot' b ≤ c ↔ a ≤ c := by
+  cases a
+  · simpa using h rfl
+  · simp [some_eq_coe]
```
Comment (YaelDillies) [inline Mathlib/Order/WithBot.lean:407]: Maybe keep the `#align` with a `ₓ`?
```suggestion
theorem unbot'_le_iff [LE α] {a : WithBot α} {b c : α} (h : a = ⊥ → b ≤ c) :
    a.unbot' b ≤ c ↔ a ≤ c := by
  cases a
  · simpa using h rfl
  · simp [some_eq_coe]
#align with_bot.unbot'_bot_le_iff WithBot.unbot'_bot_le_iffₓ
```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

File: Mathlib/Order/WithBot.lean (hunk 5 above)
Comment (jcommelin) [reply Mathlib/Order/WithBot.lean:407]: I don't think the `x`s in `#align` are that useful anymore. See also https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60.E2.82.93.60.20in.20.60.23align.60.20.235847/near/390819190

File: Mathlib/Order/WithBot.lean [hunk 6]
```diff
@@ -394,16 +400,18 @@ theorem le_coe_unbot' [Preorder α] : ∀ (a : WithBot α) (b : α), a ≤ a.unb
...
+#align with_bot.unbot'_bot_le_iff WithBot.unbot'_le_iffₓ
```
Comment (jcommelin) [inline Mathlib/Order/WithBot.lean:408]:
```suggestion
#align with_bot.unbot'_bot_le_iff WithBot.unbot'_le_iff
```

**Pull request closed without merging**